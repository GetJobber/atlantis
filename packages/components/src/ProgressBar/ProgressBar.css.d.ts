export const container: string;
export const content: string;
