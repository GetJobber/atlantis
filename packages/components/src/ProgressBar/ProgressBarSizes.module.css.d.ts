declare const styles: {
  readonly "smaller": string;
  readonly "small": string;
  readonly "base": string;
};
export = styles;

