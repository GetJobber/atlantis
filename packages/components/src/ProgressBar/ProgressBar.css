:root {
  --progress-bar--height: var(--space-base);
}

progress.ProgressBar[value] {
  width: 100%;
  height: var(--progress-bar--height);
  border: none;
  vertical-align: initial;
  appearance: none;
}

progress.ProgressBar[value],
progress.ProgressBar[value]::-webkit-progress-bar {
  box-shadow: inset 0px 2px 4px rgba(0, 0, 0, 0.05);
  border-radius: calc(var(--progress-bar--height) / 2);
}

progress.ProgressBar[value]::-webkit-progress-bar {
  background-color: var(--color-surface--background);
}

progress.ProgressBar[value]::-webkit-progress-value {
  border-radius: calc(var(--progress-bar--height) / 2);
  background-color: var(--color-interactive);
  transition: all var(--timing-slow) ease-out;
}

progress.ProgressBar[value]::-moz-progress-bar {
  border-radius: calc(var(--progress-bar--height) / 2);
  background-color: var(--color-interactive);
  transition: all var(--timing-slow) ease-out;
}
