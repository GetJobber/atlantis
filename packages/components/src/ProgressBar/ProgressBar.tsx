import React from "react";
import classnames from "classnames";
import styles from "./ProgressBar.css";
import sizes from "./Sizes.css";

interface ProgressBarProps {
  /**
   * The current step that the progress bar is on.
   */
  readonly currentStep: number;

  /**
   * The total steps to use. For percentages you can set this to 100.
   */
  readonly totalSteps: number;

  /**
   * Set the size of the progress bar
   * @default base
   */
  readonly size?: keyof typeof sizes;
}

export function ProgressBar({
  currentStep,
  totalSteps,
  size = "base",
}: ProgressBarProps) {
  const percentage = (currentStep / totalSteps) * 100;
  const progressBarClassName = classnames(styles.ProgressBar, sizes[size]);

  return (
    <progress
      className={progressBarClassName}
      max={totalSteps}
      value={currentStep}
    >
      {percentage}%
    </progress>
  );
}
