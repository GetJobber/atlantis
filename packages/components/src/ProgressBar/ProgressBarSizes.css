.small {
  --progress-bar--height: var(--space-smaller);
}

.base {
  --progress-bar--height: var(--space-base);
}
