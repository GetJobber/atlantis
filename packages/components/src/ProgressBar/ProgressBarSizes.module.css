.smaller {
  --progress-bar--height: 4px;
}

.small {
  --progress-bar--height: 8px;
}

.base {
  --progress-bar--height: 16px;
}
