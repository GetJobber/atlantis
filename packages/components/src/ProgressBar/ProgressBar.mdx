---
name: ProgressBar
menu: Components
route: /components/progress-bar
---

import { Playground, Props } from "docz";
import { ComponentStatus } from "@jobber/docx";
import { useState } from "react";
import { ProgressBar } from ".";

# ProgressBar

<ComponentStatus stage="ready" responsive="yes" accessible="yes" />

A progress bar is a visual indicator of how close something is to completion.

```ts
import { ProgressBar } from "@jobber/components/ProgressBar";
```

<Playground>
  <ProgressBar currentStep={3} totalSteps={4} />
</Playground>

## Design & Usage Guidelines

The progress bar should be used to show "definite" progress; we know exactly how
close the process is to completion. For "indefinite" progress, where we may not
know exactly how much longer something might take, use a spinner.

Some great use cases for a progress bar include:

- Setup wizard, where we know how many steps the user has completed and how many
  steps remain
- File uploads, where we know the total file size and how much data has already
  been sent

An example where you might be better served using a spinner:

- Loading a calendar within a view, where we do not know it's "complete" until
  there's no more data left to load

## Props

<Props of={ProgressBar} />

---

## Example with State

<Playground>
  {() => {
    const [step, setStep] = useState(2);
    const totalSteps = 4;
    return (
      <>
        <button onClick={() => setStep(Math.max(0, step - 1))}>
          Step Back
        </button>
        <button onClick={() => setStep(Math.min(totalSteps, step + 1))}>
          Step Forward
        </button>
        <ProgressBar currentStep={step} totalSteps={totalSteps} />
        Step: <strong>{step}</strong> of <strong>{totalSteps}</strong>
      </>
    );
  }}
</Playground>
