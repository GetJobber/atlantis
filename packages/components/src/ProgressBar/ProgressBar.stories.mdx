import { ArgsTable, Canvas, Meta, Story } from "@storybook/addon-docs";
import { useState } from "react";
import { ProgressBar } from "@jobber/components/ProgressBar";
import { Content } from "@jobber/components/Content";
import { Button } from "@jobber/components/Button";

<Meta title="Components/ProgressBar" component={ProgressBar} />

# ProgressBar

A progress bar is a visual indicator of how close something is to completion.

```ts
import { ProgressBar } from "@jobber/components/ProgressBar";
```

<Canvas>
  <Story
    name="ProgressBar"
    args={{
      currentStep: 3,
      totalSteps: 4,
    }}
  >
    {args => <ProgressBar {...args} />}
  </Story>
</Canvas>

<ArgsTable of={ProgressBar} story="ProgressBar" />

## Design & usage guidelines

The progress bar should be used to show "definite" progress; we know exactly how
close the process is to completion. For "indefinite" progress, where we may not
know exactly how much longer something might take, use a spinner.

Some great use cases for a progress bar include:

- Setup wizard, where we know how many steps the user has completed and how many
  steps remain
- File uploads, where we know the total file size and how much data has already
  been sent

An example where you might be better served using a spinner:

- Loading a calendar within a view, where we do not know it's "complete" until
  there's no more data left to load

---

## Example with state

<Canvas>
  {() => {
    const [step, setStep] = useState(2);
    const totalSteps = 4;
    return (
      <Content>
        <Button
          label="Step Back"
          onClick={() => setStep(Math.max(0, step - 1))}
        />{" "}
        <Button
          label="Step Forward"
          onClick={() => setStep(Math.min(totalSteps, step + 1))}
        />
        <ProgressBar currentStep={step} totalSteps={totalSteps} />
        Step: <strong>{step}</strong> of <strong>{totalSteps}</strong>
      </Content>
    );
  }}
</Canvas>
