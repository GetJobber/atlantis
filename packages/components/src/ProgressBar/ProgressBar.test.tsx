import React from "react";
import renderer from "react-test-renderer";
import { ProgressBar } from "./ProgressBar";

describe("with props", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<ProgressBar currentStep={2} totalSteps={10} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
