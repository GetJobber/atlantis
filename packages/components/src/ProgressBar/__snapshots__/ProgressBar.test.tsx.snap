// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with props accepts small size 1`] = `
<progress
  className="ProgressBar small"
  max={3}
  value={2}
>
  66.66666666666666
  %
</progress>
`;

exports[`with props renders correctly 1`] = `
<progress
  className="ProgressBar base"
  max={10}
  value={2}
>
  20
  %
</progress>
`;
